lib/chainlock/chainlock.js

```diff
@@ -0,0 +1,273 @@
+
+
+/**
+ * Creates ChainLock from a Buffer.
+ * @param {Buffer} buffer - A buffer of the chainLock
+ * @return {ChainLock} - An instance of ChainLock
+ */
+ChainLock.fromBuffer = function fromBuffer(buffer) {
+  return ChainLock.fromBufferReader(new BufferReader(buffer));
+};
+
+/**
+ * Create ChainLock from an object
+ * @param {Object} obj - an object with all properties of chainlock
+ * @return {ChainLock}
+ */
+ChainLock.fromObject = function fromObject(obj) {
+  var data = ChainLock._fromObject(obj);
+  return new ChainLock(data);
+};
+
+/**
+ * Verify that the signature is valid against the Quorum using quorumPublicKey
+ * @param {QuorumEntry} quorumEntry - quorum entry to test signature against
+ * @returns {Boolean} - if the signature is valid for this quorum
+ */
+ChainLock.prototype.verifySignatureAgainstQuorum = function verifySignatureAgainstQuorum(quorumEntry) {
+  return new Promise(resolve => {
+    var quorumPublicKey = quorumEntry.quorumPublicKey;
+    var signature = this.signature;
+    var signatureId = this.getSignatureIDForQuorumEntry(quorumEntry);
+
+    var blsInstance = bls.getInstance();
+    return blsInstance.then((instance) => {
+
+      var publicKey = instance.PublicKey.fromBytes(Buffer.from(quorumPublicKey, 'hex'))
+      var insecureSignature = instance.InsecureSignature.fromBytes(signature);
+      var isValid = insecureSignature.verify([signatureId], [publicKey]);
+      return resolve(isValid);
+    })
+  });
+}
+
+/**
+ * Validate Chainlock structure
+ */
+ChainLock.prototype.validate = function validate() {
+  $.checkArgument(utils.isUnsignedInteger(this.height), 'Expect height to be an unsigned integer');
+  $.checkArgument(isHexStringOfSize(this.blockHash.toString('hex'), SHA256_HASH_SIZE * 2), `Expected blockhash to be a hex string of size ${SHA256_HASH_SIZE}`);
+  $.checkArgument(isHexStringOfSize(this.signature.toString('hex'), BLS_SIGNATURE_SIZE * 2), 'Expected signature to be a bls signature');
+};
+
+/**
+ * Returns chainLock hash
+ * @returns {Buffer}
+ */
+ChainLock.prototype.getHash = function getHash() {
+  return doubleSha256(this.toBuffer()).reverse();
+};
+
+/**
+ * Computes the request ID for this ChainLock
+ * @returns {Buffer} - Request id for this chainlock
+ */
+ChainLock.prototype.getRequestId = function getRequestId() {
+  var bufferWriter = new BufferWriter();
+
+  var prefix = ChainLock.Constants.CLSIG_REQUESTID_PREFIX;
+  var prefixLength = prefix.length;
+
+  bufferWriter.writeVarintNum(prefixLength);
+  bufferWriter.write(Buffer.from(ChainLock.Constants.CLSIG_REQUESTID_PREFIX, 'utf-8'));
+  bufferWriter.writeUInt32LE(this.height);
+
+  // Double-sha is used to protect from extension attacks.
+  return doubleSha256(bufferWriter.toBuffer());
+};
+
+/**
+ * Computes signature id for a quorum entry
+ * @param {QuorumEntry} quorumEntry
+ * @returns {Buffer} - Signature id for this requestId and quorum.
+ */
+ChainLock.prototype.getSignatureIDForQuorumEntry = function (quorumEntry) {
```

Comment:
Borrowed from DashJ and Dash-Sync which seems to have put similar names. 
In here `The signature must verify against the quorum public key and SHA256(llmqType, quorumHash, SHA256(height), blockHash). llmqType and quorumHash must be taken from the quorum selected in 2.` I don't see it named. 
While message hash : ` the message hash is the block hash of the previously successful attempt.`. 

What do you advise me to do ?

Reply:
I think you can leave it as it is. I see now that it is different from message hash. It seemed confusing to me, because usually what you sign is called `message`, but I guess an ID can also be signed.